conf/machine/include/imx-base.inc

```diff
@@ -390,15 +390,18 @@ WKS_FILE_DEPENDS ?= " \
 WKS_FILE_DEPENDS_append_mx8 = " imx-boot"
 WKS_FILE_DEPENDS_append_mx8m = " imx-boot"
 
-# We need to restrict the append so we don't add this for other i.MX SoC's.
-WKS_FILE_DEPENDS_append_use-mainline-bsp_aarch64 = " imx-boot"
```

Comment:
What if someone does not change to the _boot-container override?
Also I guess this drops the depend for the i.MX 8/i.MX 8X SoC.

WKS_FILE_DEPENDS_append_use-mainline-bsp_aarch64 = "${@bb.utils.contains('OVERRIDES', 'boot-container', '', ' imx-boot', d)} "

Reply:
Additionally I guess the `WKS_FILE_DEPENDS_append_mx8m = " imx-boot"` should also get the check for boot-container override or with use-nxp-bsp imx-boot will be additionally built and probably clash somehow with the u-boot provided boot container.

Reply:
"${@bb.utils.contains('OVERRIDES', 'boot-container', '', ' imx-boot', d)} "
That will not work as OVERRIDES is a : separated variable.

Reply:
I'd change this part to `use-nxp-bsp` in _V2_, overrides are not going to fully work here.

Reply:
This has been addressed in _V2_.